shared/src/commonMain/kotlin/com/mbta/tid/mbta_app/model/TripInstantDisplay.kt

```diff
@@ -78,7 +83,31 @@ sealed class TripInstantDisplay {
             val forceAsTime = context == Context.TripDetails || scheduleBasedRouteType
             val allowTimeWithStatus =
                 context == Context.StopDetailsFiltered && routeType == RouteType.COMMUTER_RAIL
-            if (prediction?.status != null && routeType != RouteType.COMMUTER_RAIL) {
+            val showTimeAsHeadline = scheduleBasedRouteType && context != Context.TripDetails
+            val predictionTime = prediction?.stopTimeAfter(now)
+            val scheduleTime = schedule?.stopTimeAfter(now)
+            if (prediction?.status != null) {
+                if (routeType == RouteType.COMMUTER_RAIL) {
+                    when {
+                        predictionTime != null && allowTimeWithStatus ->
+                            return TimeWithStatus(
+                                predictionTime,
+                                prediction.status,
+                                showTimeAsHeadline,
+                            )
+                        predictionTime != null -> return Time(predictionTime, showTimeAsHeadline)
+                        scheduleTime != null && allowTimeWithStatus ->
```

Comment:
suggestion(non-blocking): `allowTimeWithStatus` seems like a misnomer now if there is a case where a ScheduleTimeWithStatus can be returned even if `allowTimeWithStatus` is false. 

Since it is only used in the CR case, maybe this variable can be removed entirely and the checks here could be `predictionTime != null && context = Context.StopDetailsFiltered`

Reply:
Makes sense.